ext/src/string_extension.cr

```diff
@@ -0,0 +1,76 @@
+# exact port from ActiveSupport in Ruby:
+# https://github.com/rails/rails/blob/42b0c942520e59399d70c2170253aa5275a42af1/activesupport/lib/active_support/core_ext/object/blank.rb#L101-L119
+class String
+  BLANK_RE = /\A[[:space:]]*\z/
+
+  # A string is blank if it's empty or contains whitespaces only:
+  #
+  #   ''.blank?       # => true
+  #   '   '.blank?    # => true
+  #   "\t\n\r".blank? # => true
+  #   ' blah '.blank? # => false
+  #
+  # Unicode whitespace is supported:
+  #
+  #   "\u00a0".blank? # => true
+  #
+  # @return [true, false]
+  def slow_blank?
+    BLANK_RE === self
+  end
+
+  def blank_as?
+    return true if self.nil? || self.size == 0
+    i = 0
+    while i < self.size
+      case self[i]
+      when "\u0009",
```

Comment:
Yes, I read in your great performance guide about each_char and I changed it.

Is the String iteration and comparison routines "slow"-ish?

Or is it because in the C version of fast_ruby it does:

```C
#define STR_ENC_GET(str) rb_enc_from_index(ENCODING_GET(str))
...
enc = STR_ENC_GET(str);
s = RSTRING_PTR(str);
...
e = RSTRING_END(str);
unsigned int cc = rb_enc_codepoint_len(s, e, &n, enc);
```

And with the libruby wrapper I am actually doing this to get the ruby string:

```ruby
class String
  ...
  def self.from_ruby(str : LibRuby::VALUE)
    rb_str = LibRuby.rb_str_to_str(str)
    c_str = LibRuby.rb_string_value_cstr(pointerof(rb_str))
    cr_str = String.new(c_str)
  end
end
```

Possibly dealing with cruby's `rb_str_to_str` and `rb_string_value_cstr` is different than using straight `RSTRING_PTR` in C right?

I would expect some overhead in the string casting/conversion, but the benchmarks say that the Crystal version is actually slower than the pure Ruby implementation with the regex (!!) which is what surprised me a bit. I would expect it to be slower than C but still faster than Ruby.

If you run the 'benchmark' script it will start quite the same with very small strings and as we get to strings with 60, 130 characters it becomes super slow, by orders of magnitude.

I don't think it's the blank algorithm itself but how the `string_extension_wrapper.cr` and `libruby.cr` treat the string casting. Maybe there is a better way?